### ISSUE TITLE
debugoverlay.Sphere causes instant crash on 64x servers.

### ISSUE THREAD

**Issue by wrefgtzweve** [Facepunch/garrysmod-issues#5359]
### Details
After a long struggle of debugging it turns out that `debugoverlay.Sphere` seems to instantly seg fault dedicated 64x servers. This was tested on a linux machine.

### Steps to reproduce
start 64x branch dedicated server
run `developer 1` then `lua_run debugoverlay.Sphere(Vector(), 15, 15, Color(1,1,1), true)` 
server should crash.

**Comment by brandonsturgeon** [comment]
Can confirm. This has been an issue for me for a long time.

**Comment by wrefgtzweve** [comment]
Seems like it's not a 64x issue after all, just happened on a 32x server. Seems like somehow there is something that adds a incompatibility with it despite the function being undetoured.

**Comment by robotboy655** [comment]
Cannot confirm on either main or x86-64 windows servers. Why would you even be calling debugoverlay functions on dedicated servers, they don't do anything on it?

**Comment by brandonsturgeon** [comment]
More than a few popular addons include them in shared code for easy debugging purposes

**Comment by wrefgtzweve** [comment]
I have found out why it happens, it's not 64x related it happens on 32x too, the server will instantly crash when `developer` is set to anything higher than 0 it will instantly crash the server when running it.

```
lua_run debugoverlay.Sphere(Vector(0,0,0),1,Color(1,1,1),1,1,1)
> debugoverlay.Sphere(Vector(0,0,0),1,Color(1,1,1),1,1,1)...
developer 1
lua_run debugoverlay.Sphere(Vector(0,0,0),1,Color(1,1,1),1,1,1)
> debugoverlay.Sphere(Vector(0,0,0),1,Color(1,1,1),1,1,1)...
Server marked as OFF
```

**Comment by robotboy655** [comment]
By `32x` do you mean the main branch server, or the 32bit build of the `x86-64` branch? It's a vital difference. Are you able to reproduce this on a Windows srcds? I did obviously enter `developer 1` on the server as mentioned in the first post when testing and it still didn't crash on both branches on Windows.

**Comment by wrefgtzweve** [comment]
Yeah my bad for being a bit unclear there, im using the main branch on linux. And for the 64x server we're also using linux.

**Comment by robotboy655** [comment]
Should be fixed on `dev` and in future update.
